### PR TITLE
Apply LANDING_PAGE config options for logged in users

### PR DESCRIPTION
**Pull request by schaffman5**
Allows LANDING_PAGE config variable options for logged in users and extends these options to include "home", "explore", and now "organizations".  Addresses issue #2893.

### CODE REVIEW

**Comment by techknowlogick** [comment]
I think I see a downside of applying this to logged in users as well. If it is applied to them, and say for example the landing page is set to organizations, then no one would be able to see the news feed as the redirect would always happen.

**Comment by schaffman5** [comment]
@techknowlogick  You're right.  I didn't catch that the Dashboard link just redirects to LANDING_PAGE.

I'll close this PR since this needs some additional thought about how to retain the news feed.  Is there a news URL to which the Dashboard button could be explicitly linked?

**Comment by lafriks** [comment]
One option would be to redirect after successful login to landing page

**Comment by schaffman5** [comment]
@lafriks OK.  Will reopen the PR with this suggestion.  One limitation is that it will only trigger when a user first logs in and not for subsequent sessions.

File: modules/context/auth.go [hunk 1]
```diff
@@ -42,7 +42,7 @@ func Toggle(options *ToggleOptions) macaron.Handler {
 			ctx.Redirect(setting.AppSubURL + string(setting.LandingPageURL))
 			return
 		}
-
+
```
Comment (lafriks) [inline modules/context/auth.go]: Please remove this change

File: routers/user/auth.go [hunk 2]
```diff
@@ -97,7 +97,11 @@ func checkAutoLogin(ctx *context.Context) bool {
 			ctx.SetCookie("redirect_to", "", -1, setting.AppSubURL)
 			ctx.Redirect(redirectTo)
 		} else {
-			ctx.Redirect(setting.AppSubURL + "/")
+			if setting.LandingPageURL != setting.LandingPageHome {
```
Comment (lafriks) [inline routers/user/auth.go]: I think there is no need for this as this would do the same: `ctx.Redirect(setting.AppSubURL + string(setting.LandingPageURL))`

**Review by lafriks — CHANGES_REQUESTED** [review]

File: modules/context/auth.go (hunk 1 above)
Comment (schaffman5) [reply modules/context/auth.go]: Done.

File: routers/user/auth.go (hunk 2 above)
Comment (schaffman5) [reply routers/user/auth.go]: Agreed.

**Review by lafriks — APPROVED** [review]

File: conf/app.ini [hunk 3]
```diff
@@ -581,4 +581,4 @@ FILE_EXTENSIONS = .adoc,.asciidoc
 ; External command to render all matching extensions
 RENDER_COMMAND = "asciidoc --out-file=- -"
 ; Input is not a standard input but a file
-IS_INPUT_FILE = false
\ No newline at end of file
+IS_INPUT_FILE = false
```
Comment (lafriks) [inline conf/app.ini]: Also revert this change as it's not related to this PR

**Comment by lafriks** [comment]
Also see CI failure (there is some spacing problem). Run `make fmt-check`

File: conf/app.ini (hunk 3 above)
Comment (schaffman5) [reply conf/app.ini]: This was an editor artifact but it's now reverted.  Every editor seems to introduce a final EOF new line character (vi, BBEdit, Atom, GithHub's built-in editor).  Finally had to use some perl to remove it: perl -pi -e 'chomp if eof' conf/app.ini.  Any reason why this file doesn't end with a new line?

**Comment by lafriks** [comment]
LGTM

**Comment by lunny** [comment]
LGTM
